example/convolution1D/src/convolution1D.cpp

```diff
@@ -0,0 +1,200 @@
+
+        // Since the kernel is launched 1-D calculating linearizedGlobalThreadIdx line is unnecessary.
+        // globalThreadIdx[0] can be used to map all the threads.
+        if(globalThreadIdx[0] < static_cast<unsigned>(inputSize))
+        {
+            uint32_t const halfFilterSize = filterSize / 2;
+            TElem result = 0.0f;
+            // Calculate sum of multiplications of corresponding elements
+            for(size_t i = 0; i < filterSize; ++i)
+            {
+                uint32_t inputIndex = globalThreadIdx[0] - halfFilterSize + i;
+                if(inputIndex >= 0 && inputIndex < inputSize)
+                {
+                    result += input[inputIndex] * filter[i];
+                }
+            }
+            output[globalThreadIdx[0]] = result;
+        }
+    }
+};
+
+/**
+ * @brief Function template to compare floating point values
+ *
+ */
+template<typename FloatingType>
+typename std::enable_if<std::is_floating_point<FloatingType>::value, bool>::type FuzzyEqual(
+    FloatingType a,
+    FloatingType b)
+{
+    return std::fabs(a - b) < std::numeric_limits<FloatingType>::epsilon() * 10;
+}
+
+auto main() -> int
+{
+// Fallback for the CI with disabled sequential backend
+#if defined(ALPAKA_CI) && !defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+    return EXIT_SUCCESS;
+#else
+    // Size of 1D arrays to be used in convolution integral
+    // Here instead of "convolution kernel" the term "filter" is used because kernel has a different meaning in GPU
+    // programming. Secondly filter array is not reversed. Implemented like a convolutional layer in CNN.
+    constexpr size_t filterSize = 3;
+    constexpr size_t inputSize = 8;
+    constexpr std::array<float, inputSize> expectedOutput = {0.8f, 1.4f, 2.0f, 2.6f, 3.2f, 3.8f, 4.4f, 2.3f};
+
+    // Define the index domain
+    using Dim = alpaka::DimInt<1u>;
+    // Index type
+    using Idx = std::size_t;
+
+    // Define the accelerator
+    using DevAcc = alpaka::ExampleDefaultAcc<Dim, Idx>;
+    using QueueProperty = alpaka::Blocking;
+    using QueueAcc = alpaka::Queue<DevAcc, QueueProperty>;
+    using DataType = float;
+    using BufAcc = alpaka::Buf<DevAcc, DataType, Dim, Idx>;
+
+    std::cout << "Using alpaka accelerator: " << alpaka::getAccName<DevAcc>() << std::endl;
+
+    auto const platformHost = alpaka::PlatformCpu{};
+    auto const devHost = alpaka::getDevByIdx(platformHost, 0);
+
+    // Select a device
+    auto const platformAcc = alpaka::Platform<DevAcc>{};
+    auto const devAcc = alpaka::getDevByIdx(platformAcc, 0);
+
+    // Create a queue on the device
+    QueueAcc queue(devAcc);
+
+    // Allocate memory host input
+    auto hostInputMemory = alpaka::allocBuf<DataType, Idx>(devHost, inputSize);
+    DataType* nativeHostInputMemory = alpaka::getPtrNative(hostInputMemory);
+
+    // Fill array with data
+    for(size_t i = 0; i < inputSize; i++)
+        nativeHostInputMemory[i] = static_cast<DataType>(i + 1);
```

Comment:
I think we can directly access host buffers here:
```suggestion
    auto hostInputMemory = alpaka::allocBuf<DataType, Idx>(devHost, inputSize);

    // Fill array with data
    for(size_t i = 0; i < inputSize; i++)
        hostInputMemory[i] = static_cast<DataType>(i + 1);
```
Applies to other places as well.

Reply:
Great, thanks.